03_letters/alanhillgemann/main.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprintln(out, strings.Join(sortLetters(letters("aba")), "\n"))
+}
+
+func letters(s string) map[rune]int {
+	runeMap := make(map[rune]int)
+	runeSlice := []rune(s)
+	for i := 0; i <= len(runeSlice)-1; i++ {
+		runeMap[runeSlice[i]]++
+	}
+	return runeMap
+}
+
+func sortLetters(m map[rune]int) []string {
+	stringSlice := make([]string, 0)
+	mapKeys := make([]string, 0)
+	mapValues := make([]int, 0)
+	for k, v := range m {
+		mapKeys = append(mapKeys, string(k))
+		mapValues = append(mapValues, v)
+	}
+	for i := 0; i <= len(m)-1; i++ {
```

Comment:
Use `<` to avoid the `-1`.

Reply:
Fixed. For statement consolidated into parents for range loop to make the function simpler.